comp/core/flare/helpers/builder_test.go

```diff
@@ -239,3 +241,28 @@ func TestPrepareFilePath(t *testing.T) {
 	fb.PrepareFilePath(FromSlash("a/b/c/d/file"))
 	assert.DirExists(t, filepath.Join(fb.flareDir, "a", "b", "c", "d"))
 }
+
+func TestRegisterDirPerm(t *testing.T) {
+	fb := getNewBuilder(t)
+	defer fb.clean()
+
+	root := setupDirWithData(t)
+
+	fb.RegisterDirPerm(root)
+
+	expectedPaths := common.StringSet{
+		fmt.Sprintf("%s", root):                     struct{}{},
+		fmt.Sprintf("%s/test1", root):               struct{}{},
+		fmt.Sprintf("%s/test2", root):               struct{}{},
+		fmt.Sprintf("%s/depth1", root):              struct{}{},
+		fmt.Sprintf("%s/depth1/test3", root):        struct{}{},
+		fmt.Sprintf("%s/depth1/depth2", root):       struct{}{},
+		fmt.Sprintf("%s/depth1/depth2/test4", root): struct{}{},
+	}
+
+	require.Len(t, fb.permsInfos, len(expectedPaths))
+	for path := range expectedPaths {
+		_, exist := fb.permsInfos[path]
+		require.True(t, exist)
```

Comment:
The `Contains` method should do the trick. Also we should use `assert` since we don't want to stop at the first missing item.
```suggestion
		assert.Contains(t, fb.permsInfos, path)
```